backend/controllers/activeVolunteersController.js

```diff
@@ -0,0 +1,89 @@
+import { ActiveVolunteers } from "../models/activeVolunteers.js";
+
+//query active volunteers by program, veteran, or get all active volunteers
+export const queryActiveVolunteers = async (req, res) => {
+  try {
+    const { program, veteran } = req.query;
+    let query = {};
+
+    if (program) {
+      const programsArray = program.split(",").map((p) => p.trim());
+      query.assignedProgram = { $in: programsArray };
+    }
+
+    if (veteran) {
+      query.assignedVeteran = veteran;
+    }
+
+    const filteredVolunteers = await ActiveVolunteers.find(query)
```

Comment:
Not sure if this is the correct behavior, but when testing the feature, I tried assigning user3 (veteran) to user3 user3 (volunteer). In the frontend it says user3 user3 has 0 veterans, but when I tried assigning the volunteer to the veteran, I got an error message saying "Ensure volunteer is not already assigned to veteran", despite the volunteer having 0 veterans. Is this behavior correct? 

<img width="140" alt="image" src="https://github.com/user-attachments/assets/de30529a-e933-455c-bb3f-23f3d3c9a0f5" />